src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -0,0 +1,1061 @@
+    pub rootless: bool,
+
+    /// Disable seccomp.
+    #[serde(default)]
+    pub disable_seccomp: bool,
+
+    /// Enable confidential guest support.
+    ///
+    /// Toggling that setting may trigger different hardware features, ranging from memory
+    /// encryption to both memory and CPU-state encryption and integrity.The Kata Containers
+    /// runtime dynamically detects the available feature set and aims at enabling the largest
+    /// possible one.
+    #[serde(default)]
+    pub confidential_guest: bool,
+
+    /// Path to OCI hook binaries in the *guest rootfs*.
+    ///
+    /// This does not affect host-side hooks which must instead be added to the OCI spec passed to
+    /// the runtime.
+    ///
+    /// You can create a rootfs with hooks by customizing the osbuilder scripts:
+    /// https://github.com/kata-containers/kata-containers/tree/main/tools/osbuilder
+    ///
+    /// Hooks must be stored in a subdirectory of guest_hook_path according to their hook type,
+    /// i.e. "guest_hook_path/{prestart,poststart,poststop}". The agent will scan these directories
+    /// for executable files and add them, in lexicographical order, to the lifecycle of the guest
+    /// container.
+    ///
+    /// Hooks are executed in the runtime namespace of the guest. See the official documentation:
+    /// https://github.com/opencontainers/runtime-spec/blob/v1.0.1/config.md#posix-platform-hooks
+    ///
+    /// Warnings will be logged if any error is encountered while scanning for hooks, but it will
+    /// not abort container execution.
+    #[serde(default)]
+    pub guest_hook_path: String,
+
+    /// List of valid annotation names for the hypervisor.
+    ///
+    /// Each member of the list is a regular expression, which is the base name of the annotation,
+    /// e.g. "path" for io.katacontainers.config.hypervisor.path"
+    #[serde(default)]
+    pub enable_annotations: Vec<String>,
+}
+
+impl SecurityInfo {
+    /// Adjust the configuration information after loading from configuration file.
+    pub fn adjust_configuration(&mut self) -> Result<()> {
+        if self.guest_hook_path.is_empty() {
+            self.guest_hook_path = default::DEFAULT_GUEST_HOOK_PATH.to_string();
+        }
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    pub fn validate(&self) -> Result<()> {
+        Ok(())
+    }
+
+    /// Check whether annotation key is enabled or not.
+    pub fn is_annotation_enabled(&self, path: &str) -> bool {
+        if !path.starts_with(KATA_ANNO_CONF_HYPERVISOR_PREFIX) {
+            return false;
+        }
+        let pos = KATA_ANNO_CONF_HYPERVISOR_PREFIX.len();
+        let key = &path[pos..];
+        if let Ok(set) = RegexSet::new(&self.enable_annotations) {
+            return set.is_match(key);
+        }
+        false
+    }
+
+    /// Validate path
+    pub fn validate_path(&self, path: &str) -> Result<()> {
+        validate_path!(path, "path {} is invalid{}")?;
```

Comment:
Same question about `validate` vs. `validate_…`

Reply:
validate_path will be used to validate the path from annotation